overlord/snapstate/snapstate.go

```diff
@@ -2116,6 +2116,18 @@ func doUpdate(st *state.State, requested []string, updates []update, opts Option
 			continue
 		}
 
+		if up.SnapState.RefreshFailures != nil {
```

Comment:
Could we do this in a similar manner like we track the actual failure with `.AddChangeStatusChangedHandler` - I like the "middleware" approach of that.

CC @pedronis - I don't know this area very well but it seems cleaner than a specific targetted code alterations in areas that are far away from each other.

Reply:
moved state modification under `checkSnapRefreshFailures` similar to `softCheckNothingRunningForRefresh`